pkg/engine/context/loaders/globalcontext.go

```diff
@@ -137,11 +137,14 @@ func (g *gctxLoader) loadGctxData() ([]byte, error) {
 	return json.Marshal(results)
 }
 
-func (a *gctxLoader) applyJMESPathJSON(jmesPath string, jsonData []byte) (interface{}, error) {
-	var data interface{}
-	err := json.Unmarshal(jsonData, &data)
-	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal JSON: %s, error: %w", string(jsonData), err)
+func (a *gctxLoader) applyJMESPathJSON(jmesPath string, data any) (interface{}, error) {
+	if _, ok := data.([]byte); ok {
+		var obj interface{}
+		err := json.Unmarshal(data.([]byte), &obj)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal JSON: %s, error: %w", string(data.([]byte)), err)
+		}
+		return a.jp.Search(jmesPath, obj)
 	}
 	return a.jp.Search(jmesPath, data)
```

Comment:
Is it safe to assume the data type is an interface here?

Reply:
No, it doesn't work...

Reply:
It seems that we must use json.Marshal and Unmarshal convent list result to interface{}.

Reply:
Have you profiled memory usage without this change in `applyJMESPathJSON`?

Reply:
```
Showing nodes accounting for 21622.47MB, 98.83% of 21878.05MB total
Dropped 257 nodes (cum <= 109.39MB)
Showing top 20 nodes out of 67
      flat  flat%   sum%        cum   cum%
 7351.69MB 33.60% 33.60%  7351.69MB 33.60%  bytes.growSlice
 6186.70MB 28.28% 61.88%  9837.07MB 44.96%  encoding/json.(*decodeState).objectInterface
 3964.50MB 18.12% 80.00% 11538.72MB 52.74%  encoding/json.Marshal
 2468.70MB 11.28% 91.29%  2514.20MB 11.49%  encoding/json.unquote (inline)
  509.51MB  2.33% 93.61%  2276.69MB 10.41%  encoding/json.(*decodeState).literalInterface
  440.25MB  2.01% 95.63%  9839.16MB 44.97%  encoding/json.(*decodeState).arrayInterface
  281.57MB  1.29% 96.91%   462.59MB  2.11%  sigs.k8s.io/json/internal/golang/encoding/json.(*decodeState).objectInterface
  130.51MB   0.6% 97.51%   131.51MB   0.6%  sigs.k8s.io/json/internal/golang/encoding/json.unquote (inline)
     123MB  0.56% 98.07%  9837.07MB 44.96%  encoding/json.(*decodeState).valueInterface
  121.03MB  0.55% 98.63%   223.05MB  1.02%  encoding/json.mapEncoder.encode
   24.50MB  0.11% 98.74%   116.01MB  0.53%  sigs.k8s.io/json/internal/golang/encoding/json.(*decodeState).literalInterface
   14.51MB 0.066% 98.80%   406.08MB  1.86%  sigs.k8s.io/json/internal/golang/encoding/json.(*decodeState).arrayInterface
    5.50MB 0.025% 98.83%   455.59MB  2.08%  sigs.k8s.io/json/internal/golang/encoding/json.(*decodeState).valueInterface
    0.50MB 0.0023% 98.83%   464.09MB  2.12%  k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.unstructuredJSONScheme.decodeToUnstructured
         0     0% 98.83%     7184MB 32.84%  bytes.(*Buffer).Grow (inline)
         0     0% 98.83%   167.69MB  0.77%  bytes.(*Buffer).Write
         0     0% 98.83%  7351.69MB 33.60%  bytes.(*Buffer).grow
         0     0% 98.83%   390.22MB  1.78%  encoding/json.(*Encoder).Encode
         0     0% 98.83%  9839.16MB 44.97%  encoding/json.(*decodeState).array
         0     0% 98.83%  9839.16MB 44.97%  encoding/json.(*decodeState).unmarshal
```

Reply:
It will call json.Marshal&Unmarshal in every `loadGctxData`.

Reply:
There's no difference in memory usage without updating `applyJMESPathJSON`. Can we conclude that this method consumes the most memory?

Reply:
https://github.com/kyverno/kyverno/blob/v1.13.1/pkg/engine/context/loaders/globalcontext.go#L107
https://github.com/kyverno/kyverno/blob/v1.13.1/pkg/engine/context/loaders/globalcontext.go#L142

Reply:
<img width="1369" alt="image" src="https://github.com/user-attachments/assets/c91ede3e-49d9-4f70-9c57-d15e3e487648">
`applyJMESPathJSON ` uses about 45% of memory.

Reply:
Well if JP search only works with the interface I'm not sure how much we could improve here 🤔

Reply:
Me too, but if this problem cannot be optimized, I think GlobalContextEntry is not suitable for large-scale cluster scenarios for the time being. In our production environment, I have temporarily removed the GlobalContextEntry and used direct requests to the apiserver instead.

Reply:
<img width="1817" alt="image" src="https://github.com/user-attachments/assets/cf6be650-cdb4-48e2-a37a-e9b662b99fa1">

We have a 5000 nodes k8s cluster, we use GlobalContextEntry to cache all nodes, kyverno still OOM even if the memory limit has been adjusted to 64G.

Reply:
Thanks @liuxu623 for sharing your use case, we could build a TTL cache to store resolved variables.

I have added this topic to the [next maintainers meeting](https://docs.google.com/document/d/1I_GWsz32gLw8sQyuu_Wv0-WQrtRLjn9FuX2KGNkvUY4/edit?tab=t.0).